flake.nix

```diff
@@ -110,11 +110,28 @@
         };
         default = self.packages.${system}.jujutsu;
       };
+
       apps.default = {
         type = "app";
         program = "${self.packages.${system}.jujutsu}/bin/jj";
       };
+
       formatter = pkgs.nixpkgs-fmt;
+
+      checks.jujutsu = self.packages.${system}.jujutsu.overrideAttrs ({ ... }: {
+        # FIXME (aseipp): when running `nix flake check`, this will override the
+        # main package, and nerf the build and installation phases. this is
+        # because for some inexplicable reason, the cargo cache gets invalidated
+        # in between buildPhase and checkPhase, causing every nix CI build to be
+        # 2x as long.
```

Comment:
Is there an upstream issue we can cite here?

Reply:
No, I looked around for a while but couldn't see anything conclusive or find any prior bugs that seemed related. But I can also file a bug and see if anyone else has ideas about why this is happening.

Reply:
That'll be helpful for discovering an upstream fix in the future.

Reply:
I've filed https://github.com/NixOS/nixpkgs/issues/291222 and added a reference to it in the comments.